sound/soc/intel/boards/sof_maxim_common.c

```diff
@@ -62,41 +62,35 @@ static int max98373_hw_params(struct snd_pcm_substream *substream,
 int max98373_trigger(struct snd_pcm_substream *substream, int cmd)
 {
 	struct snd_soc_pcm_runtime *rtd = asoc_substream_to_rtd(substream);
-	struct snd_soc_dai *codec_dai;
+	struct snd_soc_dapm_context *dapm;
+	struct snd_soc_dai *cpu_dai;
 	int j;
 	int ret = 0;
 
 	/* set spk pin by playback only */
 	if (substream->stream == SNDRV_PCM_STREAM_CAPTURE)
 		return 0;
 
-	for_each_rtd_codec_dais(rtd, j, codec_dai) {
-		struct snd_soc_component *component = codec_dai->component;
-		struct snd_soc_dapm_context *dapm =
-				snd_soc_component_get_dapm(component);
-		char pin_name[MAX_98373_PIN_NAME];
-
-		snprintf(pin_name, ARRAY_SIZE(pin_name), "%s Spk",
-			 codec_dai->component->name_prefix);
-
-		switch (cmd) {
-		case SNDRV_PCM_TRIGGER_START:
-		case SNDRV_PCM_TRIGGER_RESUME:
-		case SNDRV_PCM_TRIGGER_PAUSE_RELEASE:
-			ret = snd_soc_dapm_enable_pin(dapm, pin_name);
-			if (!ret)
-				snd_soc_dapm_sync(dapm);
-			break;
-		case SNDRV_PCM_TRIGGER_STOP:
-		case SNDRV_PCM_TRIGGER_SUSPEND:
-		case SNDRV_PCM_TRIGGER_PAUSE_PUSH:
-			ret = snd_soc_dapm_disable_pin(dapm, pin_name);
-			if (!ret)
-				snd_soc_dapm_sync(dapm);
-			break;
-		default:
-			break;
-		}
+	cpu_dai = asoc_rtd_to_cpu(rtd, 0);
+	dapm = snd_soc_component_get_dapm(cpu_dai->component);
+
+	switch (cmd) {
+	case SNDRV_PCM_TRIGGER_START:
+	case SNDRV_PCM_TRIGGER_RESUME:
+	case SNDRV_PCM_TRIGGER_PAUSE_RELEASE:
+		ret = snd_soc_dapm_enable_pin(dapm, "Spk");
```

Comment:
snd_soc_dapm_enable_pin could enable two spk in one invocation ?

Reply:
> invocation

Yes, because "Spk" is now connected to both "Left BE_OUT" and "Right BE_OUT".
```
const struct snd_soc_dapm_route max_98373_dapm_routes[] = {
	/* speaker */
	{ "Spk", NULL, "Left BE_OUT" },
	{ "Spk", NULL, "Right BE_OUT" },
};
```